src/Widgets/AppContainers/AbstractAppContainer.vala

```diff
@@ -453,10 +453,18 @@ namespace AppCenter {
         }
 
         private async void uninstall_clicked () {
-            if (yield package.uninstall ()) {
-                // Remove this app from the Installed Apps View
-                MainWindow.installed_view.remove_app.begin (package);
-            }
+            package.uninstall.begin ((obj, res) => {
+                try {
+                    if (package.uninstall.end (res)) {
+                        MainWindow.installed_view.remove_app.begin (package);
+                    }
+                } catch (Error e) {
+                    // Disable error dialog for if user clicks cancel. Reason: Failed to obtain authentication
+                    if (e.domain != Pk.ClientError.quark ()) {
```

Comment:
This will disable the error dialog for most errors, I've looked into this and you need to do something more like this:
```
// Disable error dialog for if user clicks cancel. Reason: Failed to obtain authentication
// Pk ErrorEnums are mapped to the error code at an offset of 0xFF (see packagekit-glib2/pk-client.h)
if (!(e is Pk.ClientError) || e.code != Pk.ErrorEnum.NOT_AUTHORIZED + 0xFF) {
```